### ISSUE TITLE
Env Vars with single quotes(') in head or tail are truncated

### ISSUE THREAD

**Issue by eladidan** [convox/rack#2172]
Setting env vars via console or cli, that have a single quote mark at head or tail of the value, truncates the quote mark.
To reproduce:
```
convox env set "foo=bar'"
convox env get foo
bar

convox env set "foo='bar"
convox env get foo
bar
```

However, mid-word works fine:
```
convox env set "foo=ba'r"
convox env get foo
ba'r
```
As well as double quotes:
```
convox env set 'foo=bar"'
convox env get foo
bar"
```

This bug was introduced after upgrading rack from version `20170202012654` to `20170413140500`

**Comment by nzoschke** [comment]
Here's the culprit:

https://github.com/convox/rack/blob/master/api/models/environment.go#L227

This behavior is a side-effect of Heroku compatibility. It's `heroku config` outputs often wrap strings in single quotes, and you generally do not want those single quotes present in the env var value the program sees.

https://github.com/convox/rack/blob/master/api/models/environment_test.go

I can see how stripping a single quote mark is having a totally different effect.

**Comment by nzoschke** [comment]
I'm thinking we should be more careful to only strip single quotes if they are a pair at the beginning and the end of the input.

**Comment by MiguelMoll** [comment]
Going to get a quick fix for this out. Going with @nzoschke thought and only remove the single quote if both first and last characters are one. Otherwise, just remove the leading quote.

**Comment by eladidan** [comment]
wouldn't this fix still strip env vars from `'bar'` to `bar`? Seems to me tat the underlying issue is that you are applying a fix that has to do with heroku compatibility globally.

**Comment by MiguelMoll** [comment]
That's correct. It's to help users bring over their environment. Would that still affect your use case?

**Comment by eladidan** [comment]
it would not but trying to get a sense of the scope of this fix

**Comment by MiguelMoll** [comment]
Understood. A fix for this issue went out via [20170427232658](https://github.com/convox/rack/releases/tag/20170427232658). Reopen this issue if you still run into problems. 👍
